### PR TITLE
Report View Updated

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<%/*<!--
+ ~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (himasha) [inline components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag]: Year is missing.

Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag]: Fixed in commit 9cc058a725ffcb1beba792c90838c6397f7c01fe

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js [hunk 2]
```diff
@@ -498,3 +497,186 @@ function selectProcessForInstanceCount(){
...
+        success: function (data) {
+
+            var array = eval('('+data+')');
+            google.load("visualization", "1", {packages:["corechart"]});
+            google.setOnLoadCallback(drawChart(array));
+
+
+            function drawChart(data) {
+                var dataArr = [['Months', 'Completed Tasks','Tasks Started']];
+                for(var i = 0;i < data.length;i++){
```
Comment (amalhub) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Apply code formatting. (Spaces missing between operators)

**Comment by NatashaWso2** [comment]
Fixed with commit 6401754

File: components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+				response.content = arrayOfTaskCountPerUser;
+			}
+			else if( option === "taskduration"){
+				response.content = resultArray;
+			}
+			else if( option === "avgprocessduration"){
+				response.content = processBarChartAvgTime;
+			}
+			else if(option === "processinstancecount"){
+            	response.content = processBarChart;
```
Comment (milindaperera) [inline components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag]: Reformat code

Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/controller/reportController.jag]: Fixed with commit 4d59f77

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js (hunk 2 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Already fixed. 
 https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/bpmn-explorer-web/src/web/js/actions.js#L854

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag [hunk 4]
```diff
@@ -0,0 +1,429 @@
+<%/*<!--
+ ~ Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sinthuja) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: Doesn't have year in the license header.

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+        var timeForOne = timeduration.data.data[i].durationInMillis;
+        var timeInMinutes = ((timeForOne /(1000*60))%60);
+        totalTime = totalTime + timeInMinutes;
+      }
+
+   var averageTime = totalTime / count;
+   var avgRoundedTime = Math.round(averageTime * 1000) / 1000;
+
+    if(checkForDuplicates(processBarChartAvgTime,processName) == 1){
+
```
Comment (sinthuja) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: Empty if??

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+    }else{
+
```
Comment (sinthuja) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: Remove unnecessary spaces.

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag (hunk 4 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: Fixed with commit 9cc058a725ffcb1beba792c90838c6397f7c01fe

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag (hunk 5 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: This empty if condition is removed when refactoring the code in the class, so this issue is already fixed.
https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag

File: components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag (hunk 6 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag]: The unnecessary spaces has been removed when refactoring the code , so this issue is already fixed.
https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/bpmn-explorer-web/src/web/model/reportModel.jag

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js [hunk 7]
```diff
@@ -498,3 +497,186 @@ function selectProcessForInstanceCount(){
...
+        type: 'GET',
+        contentType: "application/json",
+        url: url,
+        success: function (data) {
+
+            var array = eval('('+data+')');
+            google.load("visualization", "1", {packages:["corechart"]});
+            google.setOnLoadCallback(drawChart(array));
+
+
```
Comment (jranabahu) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Remove extra new lines

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js [hunk 8]
```diff
@@ -498,3 +497,186 @@ function selectProcessForInstanceCount(){
...
```
Comment (jranabahu) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Remove extra new lines

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js (hunk 7 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Already fixed 
https://github.com/wso2/carbon-business-process/blob/c0a06bc185b6da310315d95923e7da2a52b7adb6/components/bpmn/bpmn-explorer-web/src/web/js/actions.js#L710

File: components/bpmn/bpmn-explorer-web/src/web/js/actions.js (hunk 8 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-explorer-web/src/web/js/actions.js]: Already fixed
https://github.com/wso2/carbon-business-process/blob/c0a06bc185b6da310315d95923e7da2a52b7adb6/components/bpmn/bpmn-explorer-web/src/web/js/actions.js#L781

**Pull request merged**